### PR TITLE
P4_16 Add size table property

### PR DESCRIPTION
**Pull request by jafingerhut**

### CODE REVIEW

File: p4-16/spec/P4-16-spec.mdk [hunk 1]
```diff
@@ -5277,6 +5278,34 @@ of action `a_with_control_params`. Once the program is loaded,
...
+that is capable of storing `N` table entries.  This does not guarantee
+that an _arbitrary_ set of `N` entries can always be inserted in such
+a table, only that there is _some_ set of `N` entries that can be
+inserted.  For example, attempts to add some combinations of `N`
+entries may fail because the compiler selected a hash table with
+`O(1)` guaranteed search time.  See "Size property of P4 tables and
+parser value sets"[^P4SizeProperty] for further discussion on some P4
+table implementations and what they are able to guarantee.
+
+[^P4SizeProperty]: <https://github.com/jafingerhut/p4-guide/blob/master/docs/p4-table-and-parser-value-set-sizes.md> TBD: please suggest a better place for such documents, perhaps a separate directory within <https://github.com/p4lang/p4-spec>
```
Comment (mbudiu-vmw) [inline p4-16/spec/P4-16-spec.mdk]: Yes, we should use a more official URL. A file in that folder would be fine.

**Comment by jafingerhut** [comment]
I have copied the MarkDown format document I wrote giving details on various common high-speed P4 table implementations into a new directory p4-16/spec/docs/ inside this repository, and linked to that document's URL on Github inside the proposed change to the P4_16 language spec.

If someone has another URL they would prefer that link to point at, let me know what it ought to be, and how to copy the file there.

File: p4-16/spec/P4-16-spec.mdk (hunk 1 above)
Comment (jafingerhut) [reply p4-16/spec/P4-16-spec.mdk]: I created a subdirectory p4-16/spec/docs/ and copied the file into there.

File: p4-16/spec/P4-16-spec.mdk [hunk 2]
```diff
@@ -4962,10 +4962,11 @@ The standard table properties include:
   is computed.
 - `actions`: A list of all actions that may be found in the table.
 
-In addition, the tables may optionally define the following property,
+In addition, the tables may optionally define the following properties,
 
 - `default_action`: an action to execute when the lookup in the
   lookup table fails to find a match for the key used.
+- `size`: an integer specifying the desired size of the table.
```
Comment (jnfoster) [inline p4-16/spec/P4-16-spec.mdk]: Please update with "compile-time known value"

**Comment by jnfoster** [comment]
Thanks @jafingerhut!

**Comment by mbudiu-vmw** [comment]
Can we merge this?

**Comment by jafingerhut** [comment]
I believe that was the decision made in the last language design work group meeting.  Perhaps it was expected that I should merge it, but neglected to do so.  Want me to go ahead and do it now (feel free to beat me to the button click).

**Comment by jnfoster** [comment]
👍

**Pull request merged**